### ISSUE TITLE
Add prompt to enable results sharing after "Publish Poll" validation

### ISSUE THREAD

**Issue by ggordn3r** [iambateman/RankIt#148]
Several users have been confused by the new "hide results" feature and mistakenly submitted bug reports that their results weren't showing up.

To avoid this, we should add a dialog prompt after the current "Publish Poll" validation forcing users to explicitly choose whether their results will be held or shown.

Current Message after clicking "Publish Poll":
"Are you sure? Once you publish a poll others will be able to see it and vote on it until you close the poll." OK / Cancel
After "OK": [validation message]

**New dialog box after the validation message:**
Text: "Do you want to show your results immediately after each vote, or hide them from voters until you close the poll?
Choice A: Show results
Choice B: Hide results

If user chooses "Show results"
- Set `results_public` to `True`
- Show dialog message: "Your results are visible. Voters in your poll will see the current results immediately after they cast their vote.

If user chooses "Hide results":
- Set `results_public` to `False`
- Show dialog message: "Your results are hidden. To show this poll's results later, click the toggle beside "Do not show votes publicly".

**Comment by ggordn3r** [comment]
Sorry, switched this back to me because the client mentioned they may want a different solution.

**Comment by proggeramlug** [comment]
Implemented!

**Comment by ggordn3r** [comment]
Close, but I'm seeing "Ok" and "Cancel" when the buttons should be "Show results" and "Hide results"

Is that possible, or do I need to reword the message for "Ok" and "Cancel" to make sense?

![Screenshot (391)](https://user-images.githubusercontent.com/10144772/88845832-bb996e80-d1b2-11ea-83c7-55a1813add5b.png)

**Comment by proggeramlug** [comment]
Quick solution: We can't choose the words. We could use another style of popup but I think we could just reword it too.

(other style of popup is more work to implement but it is possible)

**Comment by ggordn3r** [comment]
Ok, here's the rewording to use:

Text: Click "OK" to show your results immediately after each vote. Click "Cancel" to hide them from voters until you close the poll.

**Comment by proggeramlug** [comment]
Done

**Comment by ggordn3r** [comment]
Not quite, but close. See screenshots below.

The text I provided (_Click "OK" to show your results immediately after each vote. Click "Cancel" to hide them from voters until you close the poll._) should be on the box that appears in this screenshot (i.e. the second box):

![Screenshot (23)](https://user-images.githubusercontent.com/10144772/90023880-3ddd6480-dc82-11ea-8d21-1906ecc3f2e5.png)

For the final message (screenshot below), display this:
`If they clicked OK` _Your poll is published and ready to share. Voters will see results after they submit their vote._
`If they clicked Cancel` _Your poll is published and ready to share. Voters will not see results until/unless you choose to display them in your settings._
![Screenshot (24)](https://user-images.githubusercontent.com/10144772/90024028-6d8c6c80-dc82-11ea-8487-e271305560a5.png)

**Comment by proggeramlug** [comment]
Fixed!
